autonomi/src/client/high_level/files/fs_public.rs

```diff
@@ -23,12 +26,46 @@ impl Client {
         data_addr: &DataAddress,
         to_dest: PathBuf,
     ) -> Result<(), DownloadError> {
-        let data = self.data_get_public(data_addr).await?;
+        info!("Downloading public file to {to_dest:?} from {data_addr:?}");
+
+        // Create parent directories if needed
         if let Some(parent) = to_dest.parent() {
             tokio::fs::create_dir_all(parent).await?;
             debug!("Created parent directories {parent:?} for {to_dest:?}");
         }
-        tokio::fs::write(to_dest.clone(), data).await?;
+
+        // Get the data map chunk from the public address
+        let data_map_chunk = self
+            .chunk_get(&ChunkAddress::new(*data_addr.xorname()))
+            .await?;
+        let data_map = self.deserialize_data_map(data_map_chunk.value())?;
```

Comment:
For super large files, the old data_map nesting scheme doesn't guarantee all content_chunk_infos are included in the root data_map_chunk (here).
Hence, a recursive network fetching is mandatory for the backward compatible.

There is facility of `fetch_from_data_map_chunk` which contains the work doing almost the same, just need to separate the part out as individual to avoid duplication and fix this.